Docker/create_tarfiles.py

```diff
@@ -142,35 +205,54 @@ def pdf_to_png(pdf):
         pix.save(f"{prefix}.png")  # store image as a PNG
     return True
 
-def krona_plot(ko_result,prefix):
+
+def krona_plot(ko_result, prefix):
     ko_list = glob.glob("*.ko")
     if ko_list:
-        df = pd.read_csv(ko_result,sep="\t")
-        krona_list=[]
+        df = pd.read_csv(ko_result, sep="\t")
+        krona_list = []
         for ko in ko_list:
             krona_list.append(f"{ko}.krona")
-            df[[ko,'pathway group','name']].to_csv(f"{ko}.krona",sep="\t",index=False,header=False)
-        cmd=["ktImportText"] + krona_list + [ "-o" , f"{prefix}_ko_krona.html"]
-        proc = subprocess.Popen(shlex.split(" ".join(cmd)), shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            df[[ko, 'pathway group', 'name']].to_csv(f"{ko}.krona", sep="\t",
+                                                     index=False,
+                                                     header=False)
+        cmd = ["ktImportText"] + krona_list + ["-o", f"{prefix}_ko_krona.html"]
+        proc = subprocess.Popen(shlex.split(" ".join(cmd)), shell=False,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
         outs, errs = proc.communicate()
         if proc.returncode == 0:
             return f"{prefix}_ko_krona.html"
         else:
             print(errs.decode().rstrip(), file=sys.stderr)
             return
-    
-def create_new_zip(bin_dirs):
-    for dir in bin_dirs:
-        tar_file_name = f"{dir}.tar.gz"
-        with tarfile.open(tar_file_name, "w:gz") as tar:
-            for output_file_name in glob.glob(f"{dir}/*",recursive=True):
-                tar.add(f"{output_file_name}", arcname=output_file_name)
-		
+
+
+def create_tar_file(bin_dir):
+    tar_file_name = f"{bin_dir}.tar.gz"
+    with tarfile.open(tar_file_name, "w:gz") as tar:
+        for output_file_name in glob.glob(f"{bin_dir}/*", recursive=True):
+            tar.add(f"{output_file_name}", arcname=output_file_name)
+
+
+def create_tarfiles(bin_dirs):
```

Comment:
```suggestion
def create_tarfiles(bin_dirs, threads):
    """
    This parallelizes the creation of the tar files.
    """
    
    def error_handler(e):
        print(f"Error: {e}")

    with Pool(threads) as p:
        for bin_dir in bin_dirs:
            p.apply_async(create_tar_file, args=(bin_dir,), error_callback=error_handler)
        p.close()
        p.join()
```
Issues with this:
- `threads` undefined variable
- error callback expects a callable function

Reply:
I'll take a look.

Reply:
Okay.  I did basically the same but actually I'm raising the error.  I tried just passing raise but flake8 complained